lib/fabrik/Fabrik.js

```diff
@@ -31,12 +34,29 @@ class Fabrik {
       .then(managerConstructor => managerConstructor.load(plan));
   }
 
-  static createInstance(instance_id, service_id, plan_id) {
+  static createInstance(instance_id, service_id, plan_id, context) {
     const plan = catalog.getPlan(plan_id);
     assert.strictEqual(service_id, plan.service.id);
     return this
       .createManager(plan)
-      .then(manager => manager.createInstance(instance_id));
+      .then(manager => {
+        const instance = manager.createInstance(instance_id);
+        return Promise
```

Comment:
something still does not seem right in this block. Will need to try this hands on later in the day tomorrow.

Reply:
sure. But we don't want to definitely fork the platform manager to have docker and director related stuff further unless it is absolute necessary.

Reply:
If you argue that there should not be further bifurcation based on docker/director then instead of these new constructs of platform manager, we could have directly injected `CloudControllerClient` or `K8SControllerClient` (when its added) into directorInstance.
The idea of having these platform managers are they help in abstracting any platform specific operations within themselves. In the process if they need to communicate with BOSH/DOCKER they should be able to communicate & build the required data.

Reply:
Then BOSH/DOCKER specific operations are done both in DirectorInstance/DockerInstance and 
CfPlatformManager/K8sPlatformManager. Then I would suggest let's re-look at design of these service layer class segregation. In that case we should not call the new classes as '*PlatformManager', something else. Anyway DockerInstance-->DokcerManager methods placing are not not in-line with DirectorInstance-->DockerManager. Let's sync up.